flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/PulsarSourceBuilder.java

```diff
@@ -590,6 +593,12 @@ public PulsarSource<OUT> build() {
             }
         }
 
+        if (Objects.equals(startCursor, StartCursor.latest())) {
```

Comment:
nit: `StartCursor.latest().equals(startCursor)` is also null-safe, shorter and does not require a util class